src/develop/blend.c

```diff
@@ -219,6 +220,178 @@ int dt_develop_blendif_init_masking_profile(struct dt_dev_pixelpipe_iop_t *piece
   return 1;
 }
 
+static inline float calcBlendFactor(float val, float threshold)
+{
+    // sigmoid function
+    // result is in ]0;1] range
+    // inflexion point is at (x, y) (threshold, 0.5)
+    return 1.0f / (1.0f + dt_fast_expf(16.0f - (16.0f / threshold) * val));
+}
+static inline float sqrf(float a)
+{
+  return a * a;
+}
+
+static void fast_blur(float *const restrict src, float *const restrict out, const int width, const int height, const float sigma)
+{
+  // For a blurring sigma of 2.0f a 13x13 kernel would be optimally required but the 9x9 is by far good enough here 
+  float kernel[9][9];
+  const double temp = -2.0f * sqrf(sigma);
```

Comment:
do you adjust sigma with the current zoom level ? If you don't, edge detection will yield different results depending on scale (it will anyway, but scaling sigma with zooming level makes it a bit better).

Reply:
No i don't scale sigma here and so far i really think it shouldn't be done. We don't do edge detection btw. We have per-input pixel data available as input and just do some blur to suppress pixel artefacts.

Reply:
bullshit on my side, sorry :-( As we have scaled input we need to correct the sigma here. Will do that ...